### DISCUSSION TITLE
How to TEditor / TFileEditor "mapColor" ?

### DISCUSSION THREAD

**Discussion by paule32** [magiblot/tvision#153] [Q&A] [answered]
Hello,
how is the Color Table of the TEditor / TFileEditor, to "map" the Color's ?
I take a look to the Documentation, but don't found Informations about the Palette of TEditor - to map.
```
TColorAttr
TSyntaxFileEditor::mapColor(uchar index) noexcept {
    switch (index) {
        case 1: return {0xffff00, 0x008000};
        default: return errorAttr;
    }
}
```

**Comment by magiblot** [comment, marked as answer]
The palette layouts of each class are usually documented in the public headers, but in the case of `TEditor` they aren't. I guess they forgot about it. I hadn't noticed before; I will make sure to add it later.

But anyway, since the default `TEditor` palette is defined like this:

https://github.com/magiblot/tvision/blob/be6e64f85355c6c96b7c09174986fafb6a411c6a/source/tvision/teditor1.cpp#L179

It can be deduced that it's pointing to entries 6 and 7 of `TWindow`'s palette, which are `Scroller normal text` and `Scroller selected text`:

https://github.com/magiblot/tvision/blob/be6e64f85355c6c96b7c09174986fafb6a411c6a/include/tvision/views.h#L967-L979

Therefore, `TEditor`'s palette layout is the same as `TScroller`'s:

https://github.com/magiblot/tvision/blob/be6e64f85355c6c96b7c09174986fafb6a411c6a/include/tvision/views.h#L683-L689
